### PR TITLE
core[patch]: ChatPromptTemplate.init same as ChatPromptTemplate.from_…

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/core/langchain_core/prompts/chat.py [hunk 1]
```diff
@@ -921,6 +921,86 @@ class ChatPromptTemplate(BaseChatPromptTemplate):
...
+
+        kwargs = {
+            **dict(
+                input_variables=sorted(input_vars),
+                optional_variables=sorted(optional_variables),
+                partial_variables=partial_vars,
+            ),
+            **kwargs,
+        }
+        cast(Type[ChatPromptTemplate], super()).__init__(messages=_messages, **kwargs)
```
Comment (nfcampos) [inline libs/core/langchain_core/prompts/chat.py:1002]: Why is cast needed

File: libs/core/langchain_core/prompts/chat.py [hunk 2]
```diff
@@ -921,6 +921,86 @@ class ChatPromptTemplate(BaseChatPromptTemplate):
...
+
```
Comment (nfcampos) [inline libs/core/langchain_core/prompts/chat.py:1003]: Shouldn't from_messages now delegate to init

File: libs/core/langchain_core/prompts/chat.py [hunk 3]
```diff
@@ -921,6 +921,86 @@ class ChatPromptTemplate(BaseChatPromptTemplate):
...
+            Instantiation from mixed message formats:
+
+            .. code-block:: python
+
+                template = ChatPromptTemplate([
+                    SystemMessage(content="hello"),
+                    ("human", "Hello, how are you?"),
+                ])
+
```
Comment (eyurtsev) [inline libs/core/langchain_core/prompts/chat.py:975]: perhaps we add a ..versionadded and say that prior to that version initialization is via the classmethod or something of that sort?

File: libs/core/langchain_core/prompts/chat.py (hunk 1 above)
Comment (baskaryan) [reply libs/core/langchain_core/prompts/chat.py:1008]: mypy thinks its ``super().__init__()`` output BaseChatPromptTemplate

**Pull request merged**